qiskit_nature/problems/second_quantization/lattice/lattice/hyper_cubic_lattice.py

```diff
@@ -0,0 +1,251 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Hyper-Cubic Lattice"""
+from math import pi
+from typing import List, Tuple, Union, Optional, Sequence, Callable
+from itertools import product
+import numpy as np
+from matplotlib.axes import Axes
+from matplotlib.colors import Colormap
+from retworkx import PyGraph
+from .lattice import Lattice
+
+
+class HyperCubicLattice(Lattice):
+    """Hyper-cubic lattice in d dimension."""
+
+    def __init__(
+        self,
+        size: Tuple[int, ...],
+        edge_parameter: Union[complex, Tuple[complex, ...]] = 1.0,
+        onsite_parameter: complex = 0.0,
+        boundary_condition: Union[str, Tuple[str, ...]] = "open",
+    ) -> None:
+        """
+        Args:
+            size: Lengths of each dimension.
+            edge_parameter: Weights on the unit edges. Defaults to 1.0.
```

Comment:
Please explain the difference in behavior when passing a single number vs a tuple of numbers.